geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactoryTest.scala

```diff
@@ -0,0 +1,120 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * Portions Crown Copyright (c) 2016-2020 Dstl
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.accumulo.data
+
+import org.geotools.data.DataStoreFinder
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.accumulo.MiniCluster
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class AccumuloDataStoreFactoryTest extends Specification {
+
+  import scala.collection.JavaConverters._
+
+  // we use class name to prevent spillage between unit tests
+  lazy val catalog = s"${MiniCluster.namespace}.${getClass.getSimpleName}"
+
+  "AccumuloDataStoreFactory" should {
+
+    "create a password authenticated store" in {
+      val params = Map(
+        AccumuloDataStoreParams.InstanceIdParam.key -> MiniCluster.cluster.getInstanceName,
+        AccumuloDataStoreParams.ZookeepersParam.key -> MiniCluster.cluster.getZooKeepers,
+        AccumuloDataStoreParams.UserParam.key       -> MiniCluster.Users.root.name,
+        AccumuloDataStoreParams.PasswordParam.key   -> MiniCluster.Users.root.password,
+        AccumuloDataStoreParams.CatalogParam.key    -> catalog
+      ).asJava
+      AccumuloDataStoreFactory.canProcess(params) must beTrue
+      val ds = DataStoreFinder.getDataStore(params)
+      try {
+        ds must beAnInstanceOf[AccumuloDataStore]
+      } finally {
+        if (ds != null) {
+          ds.dispose()
+        }
+      }
+    }
+
+    "create a keytab authenticated store" in {
+      val params = Map(
+        AccumuloDataStoreParams.InstanceIdParam.key -> MiniCluster.cluster.getInstanceName,
+        AccumuloDataStoreParams.ZookeepersParam.key -> MiniCluster.cluster.getZooKeepers,
+        AccumuloDataStoreParams.UserParam.key       -> MiniCluster.Users.root.name,
+        AccumuloDataStoreParams.KeytabPathParam.key -> "/path/to/keytab",
+        AccumuloDataStoreParams.CatalogParam.key    -> catalog
+      ).asJava
+      if (AccumuloDataStoreFactory.isKerberosAvailable) {
+        AccumuloDataStoreFactory.canProcess(params) must beTrue
+      } else {
+        AccumuloDataStoreFactory.canProcess(params) must beFalse
+      }
+      // TODO test kerberos
```

Comment:
First, did we test Kerberos previously?

Second, let's add a specific ticket and reference it here.  If we had a previous test, @jrs53 would be miffed if we didn't put it back:).

Reply:
we had tests on the canProcess method, which we still have, but we didn't have any tests that actually used kerberos

Reply:
added https://geomesa.atlassian.net/browse/GEOMESA-2797

Reply:
and added a reference to it here

Reply:
Yes, MockAccumulo doesn't support Kerberos, so couldn't do any real testing. MiniAccumulo does, and someone should probably write some tests...

Reply:
Whew... just making sure we were screwing up something you had added previously!  https://geomesa.atlassian.net/browse/GEOMESA-2797 is for tracking exactly this task.